File: Tree/101. Symmetric Tree/Solution-Recursive.cpp
class Solution
{
public:
    bool isSymmetric(TreeNode *root)
    {
        return !root || helper(root->left, root->right);
    }

    bool helper(TreeNode *left, TreeNode *right)
    {
        if (!right || !left)
        {
            return right == left;
        }
        if (right->val != left->val)
            return false;

        return helper(left->left, right->right) && helper(left->right, right->left);
    }
};

File: Tree/114. Flatten Binary Tree to Linked List/Solution-recursive.go
var prev *TreeNode

// Added extra argument num in the function
// Added because prev is present above the main function so as its a pointer it would contain date of the previous execution, to avoid this an if clause is added

func flatten(root *TreeNode, num ...int) {
	if root == nil {
		return
	}
	// Added check if its the first entry to the function
	if len(num) == 0 {
		prev = nil

	}

	flatten(root.Right, 1)
	flatten(root.Left, 1)

	root.Right = prev
	root.Left = nil

	prev = root
}
